### PR TITLE
intentrace: init at 0.2.4

### PR DESCRIPTION
**Pull request by cloudripper**
Adding new package for [intentrace](https://github.com/sectordistrict/intentrace) - a prettified Linux syscall tracing tool written in Rust. Intentrace is in _beta_ and currently only supports x86-64.

## Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#linking-nixos-module-tests-to-a-package) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [24.11 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2411.section.md) (or backporting [23.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) and [24.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2405.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

### CODE REVIEW

File: pkgs/by-name/in/intentrace/package.nix [hunk 1]
```diff
@@ -0,0 +1,25 @@
...
+
+  cargoHash = "sha256-pyGcQy7p0+Vqv3Khy1hLgahcOpqnbKKmRLZcKwkvVWw=";
+
+  meta = with lib; {
+    description = "A prettified Linux syscall tracing tool";
+    homepage = "https://github.com/sectordistrict/intentrace";
+    license = licenses.mit;
+    platforms = [ "x86_64-linux" ];
+    mainProgram = "intentrace";
+    maintainers = with maintainers; [ cloudripper ];
```
Comment (FliegendeWurst) [inline pkgs/by-name/in/intentrace/package.nix:22]: You also need to add yourself to maintainer-list.nix :)

File: maintainers/maintainer-list.nix
Comment (FliegendeWurst) [inline maintainers/maintainer-list.nix:1]: Don't format this file :laughing:

File: pkgs/by-name/in/intentrace/package.nix [hunk 2]
```diff
@@ -0,0 +1,28 @@
...
+  src = fetchFromGitHub {
+    owner = "sectordistrict";
+    repo = "intentrace";
+    rev = "refs/tags/v${version}";
+    hash = "sha256-Bsis8tL2xahJT/qAFVbbd/CZ7n8KJYLPTIl1a1WHR4c=";
+  };
+
+  cargoHash = "sha256-pyGcQy7p0+Vqv3Khy1hLgahcOpqnbKKmRLZcKwkvVWw=";
+
+  meta = with lib; {
```
Comment (FliegendeWurst) [inline pkgs/by-name/in/intentrace/package.nix:20]: Better to avoid `with lib;`. Add `lib.` below

File: pkgs/by-name/in/intentrace/package.nix [hunk 3]
```diff
@@ -0,0 +1,28 @@
+{
+  fetchFromGitHub,
+  lib,
+  rustPlatform,
+}:
+
+rustPlatform.buildRustPackage rec {
```
Comment (FliegendeWurst) [inline pkgs/by-name/in/intentrace/package.nix:7]: I think the finalAttrs pattern is recommended over `rec`

**Comment by thtrf** [comment]
Please squash and create a separate commit for the entry in the maintainer list. More info: [commit conventions](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#commit-conventions)

File: pkgs/by-name/in/intentrace/package.nix (hunk 3 above)
Comment (cloudripper) [reply pkgs/by-name/in/intentrace/package.nix]: I went with '_let ... in_' pattern to replace rec. Let me know if that meets convention.

**Comment by cloudripper** [comment]
Thank you, [FliegendeWurst](https://github.com/FliegendeWurst) and [thtrf](https://github.com/thtrf) for the guidance. I have included suggested changes and squashed pkg init commit and created add maintainer commit. Let me know if there is anything else I might have missed that is needed to meet standards.

File: pkgs/by-name/in/intentrace/package.nix [hunk 4]
```diff
@@ -0,0 +1,34 @@
...
+
+  cargoHash = "sha256-pyGcQy7p0+Vqv3Khy1hLgahcOpqnbKKmRLZcKwkvVWw=";
+
+  meta = {
+    description = "Prettified Linux syscall tracing tool (like strace)";
+    homepage = "https://github.com/sectordistrict/intentrace";
+    license = lib.licenses.mit;
+    platforms = [
+      "x86_64-linux"
+      "aarch64"
```
Comment (FliegendeWurst) [inline pkgs/by-name/in/intentrace/package.nix:29]:
```suggestion
      "aarch64-linux"
```

Comment (FliegendeWurst) [reply pkgs/by-name/in/intentrace/package.nix:29]: Or just `lib.platforms.linux`

**Review by FliegendeWurst — COMMENTED** [review]
I think the two commits should be the other way around. LGTM otherwise

File: pkgs/by-name/in/intentrace/package.nix [hunk 5]
```diff
@@ -0,0 +1,31 @@
...
+    hash = "sha256-Bsis8tL2xahJT/qAFVbbd/CZ7n8KJYLPTIl1a1WHR4c=";
+  };
+
+  cargoHash = "sha256-pyGcQy7p0+Vqv3Khy1hLgahcOpqnbKKmRLZcKwkvVWw=";
+
+  meta = {
+    description = "Prettified Linux syscall tracing tool (like strace)";
+    homepage = "https://github.com/sectordistrict/intentrace";
+    license = lib.licenses.mit;
+    platforms = lib.platforms.linux;
```
Comment (FliegendeWurst) [inline pkgs/by-name/in/intentrace/package.nix:27]:
```suggestion
    platforms = [ "x86_64-linux" ];
```
It seems upstream currently hardcodes register names
```
error[E0609]: no field `rax` on type `&user_regs_struct`
   --> src/main.rs:326:43
    |
326 |     syscall.errno = errno_check(registers.rax);
    |                                           ^^^ unknown field
    |
    = note: available fields are: `regs`, `sp`, `pc`, `pstate`
```

Comment (06kellyjac) [reply pkgs/by-name/in/intentrace/package.nix:27]: That was also mentioned in the PR description. Good spot, worth fixing the nix to match

File: pkgs/by-name/in/intentrace/package.nix [hunk 6]
```diff
@@ -0,0 +1,31 @@
...
+    mainProgram = "intentrace";
+    maintainers = with lib.maintainers; [ cloudripper ];
```
Comment (06kellyjac) [inline pkgs/by-name/in/intentrace/package.nix:29]: I'm happy to join you in maintaining this if you'd like

```suggestion
    maintainers = with lib.maintainers; [ cloudripper jk ];
```

**Review by Aleksanaa — APPROVED** [review]

**Review by 06kellyjac — APPROVED** [review]
Only other thing would be to update to `0.2.5` but I'm fine either way
LGTM

**Pull request merged**